### ISSUE TITLE
SIFT CLI - Error: Update returned exit code not zero

### ISSUE THREAD

**Issue by J-a-s-o-n-P** [teamdfir/sift#501]
Hi,

When running sift-cli I get exit code not zero when the install hits "Running: sift-scripts":

>> Running: sift-scripts
Update returned exit code not zero
Error: Update returned exit code not zero
    at ChildProcess.<anonymous> (/snapshot/sift-cli/sift-cli.js:547:23)
    at ChildProcess.emit (events.js:315:20)
    at maybeClose (internal/child_process.js:1021:16)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:286:5)

/etc/os-release output:

NAME="Ubuntu"
VERSION="18.04.5 LTS (Bionic Beaver)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 18.04.5 LTS"
VERSION_ID="18.04"
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
VERSION_CODENAME=bionic
UBUNTU_CODENAME=bionic

Full output when trying to install is attached in sift-cli output. Saltstack log also attached. 

Any help is appreciated. 

[saltstack.log](https://github.com/teamdfir/sift/files/5718210/saltstack.log)
[sift-cli output.txt](https://github.com/teamdfir/sift/files/5718211/sift-cli.output.txt)

**Comment by timmyg558** [comment]
I am having the exact same issue with the small difference that mine starts at ">> Running: sift-config".

**Comment by ekristen** [comment]
Please provide logs. I’m looking Into it

**Comment by timmyg558** [comment]
Just reverted snapshot. I'll rerun. Which logs will you need me to attach to best assist you?

**Comment by digitalsleuth** [comment]
Hi @J-a-s-o-n-P and @timmyg558 , 
I've reviewed your saltstack logs and you both appear to have two different errors.

@J-a-s-o-n-P , your error appears to be connection / apt related. The errors indicate that you were unable to download packages from the apt-repo, and the retcode: 100 indicates that your apt database was not updated (sudo apt update).

By running the `sudo apt update` command and running the installer again, you should have no issues.

@timmyg558 , your issue appears to be a temporary one. The issue listed in your saltstack log was that you were unable to download densityscout from the source homepage. Now, this could be representative of either a temporary drop in connection (source could have been down, or your connection lost), or if you're installing SIFT from within an enterprise environment, your firewall may be blocking the download. Your error indicates:
```
archive_|-sift-tool-densityscout-archive_|-/usr/local/src/densityscout/densityscout_build_45_linux_|-extracted:
    __id__: sift-tool-densityscout-archive
    __run_num__: 348
    __sls__: sift.scripts.densityscout
    changes: {}
    comment: 'Error: [Errno 111] Connection refused reading https://cert.at/media/files/downloads/software/densityscout/files/densityscout_build_45_linux.zip'
    duration: 11.784
    name: https://cert.at/media/files/downloads/software/densityscout/files/densityscout_build_45_linux.zip
    result: false
    start_time: '15:17:54.143476'
```
Since this state could not complete, other states which depended on it could not complete either. You could try first to download the file directly from the URL listed in the error and see if you're able to download it. If you are, then running the CLI installer again should work. If it does not, please let us know.

**Comment by timmyg558** [comment]
@digitalsleuth Thank you for catching that since I completely missed it. Kicking myself. Modifications to my setup allowed successful install. Thanks again!

**Comment by michaelafry** [comment]
I had this as well, I was running as sudo, not root, I had a strong internet connection and had done the apt cleanup. I am using ubuntu 20.04. I tried the --pre-release command line parameter and it seems to be off and running.

**Comment by Coding4Sec** [comment]
Hi. Crazy: Same error as @J-a-s-o-n-P . Did fix as specified by @digitalsleuth then ran sudo sift install --mode=server and same issue appears. Any idea? /etc/os-release output same as @J-a-s-o-n-P . running ubuntu 18.04 LXC... on proxmox

**Comment by digitalsleuth** [comment]
Hi @Coding4Sec , any chance you can upload your saltstack.log file so we can take a look? There may be other items impacting your setup besides the issue which @J-a-s-o-n-P was having.

**Comment by digitalsleuth** [comment]
@Coding4Sec , any chance you could provide a progress update on this? If you're still experiencing the issue, please upload your saltstack.log file so we can take a look.

**Comment by Coding4Sec** [comment]
Sorry for being late. I was busy with exams.... So i dropped the ubuntu lxc and set up a an ubuntu VM on proxmox. I now run into the same error reported here: https://github.com/teamdfir/sift/issues/507

**Comment by Coding4Sec** [comment]
The CLI output:

**Comment by digitalsleuth** [comment]
Hi @Coding4Sec , can you also attach the saltstack.log file, and the output of `salt-common -V`?

**Comment by Coding4Sec** [comment]
> Hi @Coding4Sec , can you also attach the saltstack.log file, and the output of `salt-common -V`?

No success: command not found

**Comment by digitalsleuth** [comment]
Try `salt-call -V` or `salt-minion -V`. One of those should provide an output.

**Comment by Coding4Sec** [comment]
ok.
Strange, I did reboot, now my prompt changed to <username>@siftworkstation...

Here is output form `sat-call -V`

secfor@siftworkstation: ~
$ salt-call -V
Salt Version:
           Salt: 3001.4
 
Dependency Versions:
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: 2.8.1
      docker-py: Not Installed
          gitdb: Not Installed
      gitpython: Not Installed
         Jinja2: 2.10
        libgit2: Not Installed
       M2Crypto: Not Installed
           Mako: 1.0.7
   msgpack-pure: Not Installed
 msgpack-python: 0.5.6
   mysql-python: Not Installed
      pycparser: Not Installed
       pycrypto: 2.6.1
   pycryptodome: 3.4.7
         pygit2: Not Installed
         Python: 3.6.9 (default, Oct  8 2020, 12:12:24)
   python-gnupg: 0.4.1
         PyYAML: 3.12
          PyZMQ: 20.0.0
          smmap: Not Installed
        timelib: Not Installed
        Tornado: 4.5.3
            ZMQ: 4.3.3
 
System Versions:
           dist: ubuntu 18.04 Bionic Beaver
         locale: UTF-8
        machine: x86_64
        release: 5.4.0-65-generic
         system: Linux
        version: Ubuntu 18.04 Bionic Beaver

**Comment by digitalsleuth** [comment]
Can you run `ls -la` in your home directory and check to see that all of the folders there are owned by your user?

**Comment by Coding4Sec** [comment]
> Can you run `ls -la` in your home directory and check to see that all of the folders there are owned by your user?

sure, all but two are not:
.kde
and 
.cpanm
they are owned by root

**Comment by digitalsleuth** [comment]
Change those to be owned by your user (to match the rest) using `chown -R <user>:<user>` and try running your sift install again. That should do it.

**Comment by Coding4Sec** [comment]
Ok, Thanks. I will do and come back for feedback...

**Comment by Coding4Sec** [comment]
> Change those to be owned by your user (to match the rest) using `chown -R <user>:<user>` and try running your sift install again. That should do it.

Uhh, no success, same as before...

Anything else to do?

**Comment by digitalsleuth** [comment]
Can you upload the saltstack.log from this attempt?

**Comment by Coding4Sec** [comment]
> Can you upload the saltstack.log from this attempt?

Here we go
[saltstack.log](https://github.com/teamdfir/sift/files/5961583/saltstack.log)

**Comment by ekristen** [comment]
Potentially looks like some unicode errors. Anything unique about your install? Foreign language?

**Comment by Coding4Sec** [comment]
German... do we need english installation?

**Comment by digitalsleuth** [comment]
Can you share the output of `locale` here, and try running your sift install like this: `LANGUAGE=en_US.UTF-8 sudo sift install`
If that doesn't work, change LANGUAGE to LC_ALL.

**Comment by digitalsleuth** [comment]
You read my mind @ekristen

**Comment by Coding4Sec** [comment]
yes, will do so. Let me try tomorrow morning. Will come back then. Locale: all entries are de_DE.UTF-8

**Comment by Coding4Sec** [comment]
ok, tried 
>> en_US.UTF-8 sudo sift install 
returned exit code not zero....

**Comment by Coding4Sec** [comment]
ok, tried LC_ALL no success. I did a fresh VM ubuntu 18.04 in en. except timezone and layout in de. no success: exit code not zero... although sift-config throws error, the prompt changes to @siftworkstation

**Comment by digitalsleuth** [comment]
Can you post the saltstack.log for this as well? It's possible that another error may have been introduced besides the original which will help narrow things down. And can you share the exact command you're using as well?

**Comment by Coding4Sec** [comment]
1. $ LANGUAGE=en_US.UTF-8 sudo sift install
-> not zero

2. $ LANGUAGE=LC_ALL sudo sift install
and log: 
[saltstack.log](https://github.com/teamdfir/sift/files/5962030/saltstack.log)


log for try#1 isn't available anymore

**Comment by Coding4Sec** [comment]
this is from 2nd VM in en_US.... except timezone and keyb.-
[saltstack.log](https://github.com/teamdfir/sift/files/5962137/saltstack.log)
layout

**Comment by digitalsleuth** [comment]
Just to clarify, the second attempt should have been `LC_ALL=en_US.UTF-8 sudo sift install`> I'm looking at the log now.

**Comment by Coding4Sec** [comment]
uupps, ok, running this command... last thing to do for today... will upload log

**Comment by digitalsleuth** [comment]
Actually your last install worked but with one separate error with pyhindsight. I'll take a look at this tonight, but I'm pretty sure I know the issue, related to setuptools_rust.

**Comment by Coding4Sec** [comment]
ok, this is for 
` LC_ALL=en_US.UTF-8 sudo sift install
`
[saltstack.log](https://github.com/teamdfir/sift/files/5962244/saltstack.log)

**Comment by digitalsleuth** [comment]
That one installed fine except for pyhindsight due to the setuptools_rust error I mentioned earlier. I'll submit a pull for that tonight to fix it.

**Comment by digitalsleuth** [comment]
Issue found and resolved, but identifies a larger issue. Requires setuptools_rust and a newer pip version. I have the necessary changes ready for a pull request, but I'm in the midst of testing in both a Bionic and Focal docker (for a full fresh install), and a VM for an upgrade scenario. You can expect a pull request before EOD tomorrow.

**Comment by Coding4Sec** [comment]
@digitalsleuth Any updates on how to proceed?

**Comment by digitalsleuth** [comment]
Well the easiest way would be first to upgrade pip using `sudo python3 -m pip install --upgrade pip`, then install setuptools_rust with `sudo python3 -m pip install setuptools_rust`, and finally run `sudo python3 -m pip install pyhindsight`. 

This will complete your latest install and, once the pull request is approved, you can try your install again.

**Comment by p57p57** [comment]
Hi gents. same case as the others.
I am running a pretty old version of sift ws but with the latest cli and a brand new ubuntu 18.04.
I did the apt updates and so forth. I ran sift update and sift upgrade that both failed.
 attached the log from the upgrade.

sift version ( i have updated the cli to 1.9.2 just before the upgrade and just after i am back in 1.5.1)
> sift-cli@1.5.1-beta.0-master.154cb2f
> sift-version: v2017.27.0


the error message:
 Running: sift-config
Update returned exit code not zero
Error: Update returned exit code not zero
    at ChildProcess.update.on (/snapshot/sift-cli/sift-cli.js:488:23)
    at emitTwo (events.js:106:13)
    at ChildProcess.emit (events.js:194:7)
    at maybeClose (internal/child_process.js:899:16)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:226:5)

[saltstack.log](https://github.com/teamdfir/sift/files/6031858/saltstack.log)

**Comment by codifiedlikenessutility1970** [comment]
I am having trouble with my install dropping while running: sift-config-tools....it also changes my hwclock even doing the update this: `LC_ALL=en_US.UTF-4 sift update `

I am including my output for `salt-call -V`
 salt-call -V
Salt Version:
           Salt: 3001.7
 
Dependency Versions:
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: 2.8.1
      docker-py: Not Installed
          gitdb: Not Installed
      gitpython: Not Installed
         Jinja2: 2.10
        libgit2: Not Installed
       M2Crypto: Not Installed
           Mako: 1.0.7
   msgpack-pure: Not Installed
 msgpack-python: 0.5.6
   mysql-python: Not Installed
      pycparser: Not Installed
       pycrypto: Not Installed
   pycryptodome: 3.10.1
         pygit2: Not Installed
         Python: 3.6.9 (default, Jan 26 2021, 15:33:00)
   python-gnupg: 0.4.1
         PyYAML: 3.12
          PyZMQ: 22.0.3
          smmap: Not Installed
        timelib: Not Installed
        Tornado: 4.5.3
            ZMQ: 4.3.4
 
System Versions:
           dist: ubuntu 18.04 Bionic Beaver
         locale: UTF-8
        machine: x86_64
        release: 4.15.0-143-generic
         system: Linux
        version: Ubuntu 18.04 Bionic Beaver

and my saltstack log
[saltstack.log](https://github.com/teamdfir/sift/files/6354128/saltstack.log)

**Comment by codifiedlikenessutility1970** [comment]
This is where it drops:

>> Running: /etc/foremost.conf
>> Running: /usr/local/etc/foremost.conf
>> Running: sift-config-tools
Update returned exit code not zero
Error: Update returned exit code not zero
    at ChildProcess.<anonymous> (/snapshot/sift-cli/sift-cli.js:542:23)
    at ChildProcess.emit (events.js:315:20)
    at maybeClose (internal/child_process.js:1051:16)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:287:5)


----- PLEASE READ ----------------------

A lot of failures are caused by the apt system being locked
or unhealthy. 

Before opening an issue in GitHub, please check to see if 
your apt system is healthy.

Try running 'apt-get update' then remove any packages that 
aren't used by running 'apt-get autoremove'

**Comment by digitalsleuth** [comment]
Hi @codifiedlikenessutility1970 , can you check the permissions on the /home/timelord/.dbus directory, and confirm that it has the proper permissions (should be set to timelord:timelord or permissions specific to the timelord user)?
The error is coming from the dconf command, which is modifying the contents of the .dbus dir.

**Comment by codifiedlikenessutility1970** [comment]
Hmmm. It's root.

**Comment by codifiedlikenessutility1970** [comment]
I started the dl/install process in timelord.

**Comment by digitalsleuth** [comment]
If you haven't already, please change the permissions to the proper permissions and try running the install command again.

**Comment by ekristen** [comment]
I’ll review permissions tonight. I just did a fresh install though with no problems.

**Comment by codifiedlikenessutility1970** [comment]
Thanks, @digitalsleuth! That did it!!!....two things that were odd, but I don't think were related to this was it switching my timezone and it also moved my dock from the left side to the bottom...both happened at the end of the where it kept cutting out. Thanks again!!!

**Comment by Nima-pw** [comment]
Hi,

Dear @digitalsleuth, I have the same issue (exit code, not zero). Could you please check my logs?

Ubuntu 18:
[saltstack-18.log](https://github.com/teamdfir/sift/files/6499243/saltstack-18.log)

Ubuntu 20:
[saltstack-20.log](https://github.com/teamdfir/sift/files/6499246/saltstack-20.log)

**Comment by digitalsleuth** [comment]
Hi @Nima-pw, It looks like the issue is with pefile. I encountered this recently with REMnux, and we created a temporary fix while we await the permanent fix being merged by the original author. The issue is covered [here](https://github.com/erocarrera/pefile/issues/318).

I'll submit a pull request to change the state temporarily, and once the issue is fixed by the author, return it to its original state.

**Comment by Nima-pw** [comment]
Thanks, @digitalsleuth, for your reply.
Unfortunately, I did not fully understand the temporary solution.
The temp fix is this?
"modifying lines 3789 and 3790 to replace the f' string formatting with .format() then everything works as it should."

Where should this edit be done?

**Comment by digitalsleuth** [comment]
Hi @Nima-pw, the reference shows you where the issue was caused, I am submitting a fix shortly for this. Once merged, the issue will be resolved.

**Comment by digitalsleuth** [comment]
Pull request [53](https://github.com/teamdfir/sift-saltstack/pull/53) has been submitted to provide a temporary fix. For the time being, you may be able to install pefile manually first, then running the sift-cli installer to complete the install.

To install the 'temporary fix' version of pefile, you can run the following:
`python2 -m pip install git+https://github.com/digitalsleuth/pefile.git`

Then run the sift installer.

**Comment by Nima-pw** [comment]
Thanks, @digitalsleuth,
I will do it and come back for feedback.

**Comment by Nima-pw** [comment]
The result of installing the 'temporary fix' version:

```
Requirement already satisfied: future in /usr/local/lib/python2.7/dist-packages (from pefile==2021.5.13) (0.18.2)
Building wheels for collected packages: pefile
  Building wheel for pefile (setup.py) ... done
  Created wheel for pefile: filename=pefile-2021.5.13-py2-none-any.whl size=62568 sha256=5ce9b7fb8f0f03a1070a01e4c69e4512547d07b170ce03c77db1a383299c7da9
  Stored in directory: /tmp/pip-ephem-wheel-cache-4lWkzU/wheels/22/58/a0/33128d215f3adebb7b7d035f4d7a06f567bfa793af8b1cc630
Successfully built pefile
Installing collected packages: pefile
Successfully installed pefile-2021.5.13
```

But the saltstack.log (ubuntu 20):

Update returned exit code not zero
Error: Update returned exit code not zero
    at ChildProcess.<anonymous> (/snapshot/sift-cli/sift-cli.js:547:23)
    at ChildProcess.emit (events.js:315:20)
    at maybeClose (internal/child_process.js:1021:16)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:286:5)


----- PLEASE READ ----------------------

A lot of failures are caused by the apt system being locked
or unhealthy. 


[saltstack.log](https://github.com/teamdfir/sift/files/6502462/saltstack.log)

**Comment by digitalsleuth** [comment]
@Nima-pw , the errors seen in the saltstack.log you provided indicate that there was an issue downloading some files, not related to pefile:

`# [ERROR   ] Error: HTTP 599: Timeout while connecting reading https://didierstevens.com/files/software/virustotal-submit_V0_0_3.zip`
`# [ERROR   ] Error: HTTP 599: Timeout while connecting reading https://didierstevens.com/files/software/virustotal-search_V0_1_4.zip`

You can try reaching these sites directly to see if there is an issue with your internet connectivity or if these files are specifically blocked by your firewall. Since these are the only errors in the log, it's likely that once this connection issue is resolved, you won't have any further problems with the installation.

**Comment by Nima-pw** [comment]
@digitalsleuth you are great!
Thank you, man!

**Comment by Nima-pw** [comment]
@digitalsleuth you are great!
Thank you, man!

OMPLETED SUCCESSFULLY -- Success: 654, Failure: 0

**Comment by TylerTenlen** [comment]
@digitalsleuth  Thank you so much for the fix regarding pefiile. I was having issues with this for a few days and this resolved it.

**Comment by jackdyson1399** [comment]
Hello,
My SIFT install is failing too, with the below message:

>> Running: /etc/foremost.conf
>> Running: /usr/local/etc/foremost.conf
>> Running: sift-config-tools
>> Running: sift-config
> Update returned exit code not zero
> Error: Update returned exit code not zero
>     at ChildProcess.<anonymous> (/snapshot/sift-cli/sift-cli.js:542:23)
>     at ChildProcess.emit (events.js:315:20)
>     at maybeClose (internal/child_process.js:1051:16)
>     at Process.ChildProcess._handle.onexit (internal/child_process.js:287:5)
> ----- PLEASE READ ----------------------
> 
> A lot of failures are caused by the apt system being locked
> or unhealthy. 
> 
> Before opening an issue in GitHub, please check to see if 
> your apt system is healthy.
> 
> Try running 'apt-get update' then remove any packages that 
> aren't used by running 'apt-get autoremove'


I am attempting to install `SIFT v2021.4.4` using `sift-cli v1.10.0` on an `Ubuntu 20.04` VirtualBox VM.
The exact command I am running for the install is: `sudo sift install --version=v2021.4.4 --user=test` 
I am doing it this way as my aim is to script the SIFT install with some other components, but the script will be run with sudo privileges, hence me using sudo for the SIFT install while testing this new version.

I have attached the saltstack.log of the failed attempt [here](https://github.com/teamdfir/sift/files/6514748/saltstack-PREfix.log).


I also attempted the pefile fix mentioned above and it failed again. 
I have attached the saltstack.log for that attempt [here](https://github.com/teamdfir/sift/files/6514752/saltstack-POSTfix.log).

Hope you can help. 
I am aware that the install is mainly complete, but it would be great to see the 'SUCCESS' message and have SIFT displaying as installed rather than 'notinstalled' for my work's strict versioning control.

**Comment by digitalsleuth** [comment]
Hi @jackdyson1399 , looking at your log files, the first one shows that you had a connectivity issue, which is why it errored out (the one called PREfix). The one called POSTfix is failing due to the same pefile issue we mentioned earlier, and can be fixed temporarily by following the steps [here](https://github.com/teamdfir/sift/issues/501#issuecomment-843202682) prior to running the sift-cli.

A pull request has been merged which will see this issue fixed in the next release, but for the time being, you should be able to install pefile manually (the link above) and then SIFT will complete its install.

**Comment by digitalsleuth** [comment]
@jackdyson1399 , you'll then need to install python2 first. Use the following commands to do that, then do the pefile fix above. Make sure, as well, when you run the pefile fix that you run it with `sudo`:

`sudo apt-get install python2 git curl -y`
`curl -o /tmp/get-pip.py https://bootstrap.pypa.io/pip/2.7/get-pip.py`
`sudo python2 /tmp/get-pip.py`
`sudo python2 -m pip install git+https://github.com/digitalsleuth/pefile.git`

**Comment by jackdyson1399** [comment]
I will give this a try now and report back to you.

**Comment by ekristen** [comment]
I need to go to new release for the CLI to pick up the PE file fix.￼

However if you are wanting to programmatically install sift I would suggest you look at using salt directly in the salt stack repository versus trying to use the CLI￼

**Comment by jackdyson1399** [comment]
Just completed the SIFT update now. Thank you for your help!
I recon my issue was I did not run your pefile fix with sudo.

Interestingly though, the Desktop and the tool bar has not changed, but it did when the install failed. This is only minor but don't want to be missing anything. The saltstack.log for the successful attempt without the correct Desktop wallpaper and reorientation of the tool bar is [here](https://github.com/teamdfir/sift/files/6516378/saltstack-success.log). And I have attached a photo of the Desktop so you can see what I mean.  
![image](https://user-images.githubusercontent.com/46749580/118989107-ac6df200-b979-11eb-8bac-ba72e9bdd4da.png)

I have tried rebooting the VM with no luck.

@ekristen thank you for your advice, I will investigate using salt stack instead, but the script I am using is not hugely complicated, it mainly just clones some extra work specific repositories on top of the SIFT install.

**Comment by tompolth** [comment]
Hi everyone, 

I am trying to update and I am getting this error:
 Running: sift-config-tools
Update returned exit code not zero
Error: Update returned exit code not zero
    at ChildProcess.<anonymous> (/snapshot/sift-cli/sift-cli.js:545:23)
    at ChildProcess.emit (events.js:315:20)
    at maybeClose (internal/child_process.js:1021:16)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:286:5)

I have tested that the home folder has the right owner (sansforensics) as it was mentioned in a previous comment, but I get the same result. 
the saltstack log is attached here 
[saltstack.log](https://github.com/teamdfir/sift/files/6532171/saltstack.log)

I'd appreciate any help.

Thanks

**Comment by digitalsleuth** [comment]
Hi @tompolth , There are a few different errors there. The easiest one to fix is to follow the steps [here](https://github.com/teamdfir/sift/issues/501#issuecomment-843202682), as this will resolve the pefile issue found in the log.

The other errors center around the sha256 hash being different from some of the PDF's that are being downloaded. I'll take a look at those later today and see if the hashes need to be updated, but these won't affect the function of the SIFT environment. 

Give the fix above a try, then run the installer again. You can also visit the teamdfir/sift-saltstack repo and install SIFT using the instructions there for the up-to-date fix.

**Comment by tompolth** [comment]
Hi @digitalsleuth 

I am very sorry for my late answer. Work kept me very busy. Just thank you for your help, your solution fixed the problem

**Comment by ekristen** [comment]
This issue got very long, I'm going to close and lock, if you have similar issues in the future please open a new issue.
